src/NUnitTestAdapter/TestConverter.cs

```diff
@@ -199,6 +199,35 @@ private VSTestResult GetBasicResult(XmlNode resultNode)
             if (outputNode != null)
                 vsResult.Messages.Add(new TestResultMessage(TestResultMessage.StandardOutCategory, outputNode.InnerText));
 
+            var attachments = resultNode.SelectNodes("attachments/attachment");
+            if (attachments.Count > 0)
+            {
+                var attachmentSet = new AttachmentSet(new Uri(NUnitTestAdapter.ExecutorUri), "Attachments");
+                vsResult.Attachments.Add(attachmentSet);
+
+                foreach (XmlNode attachment in attachments)
+                {
+                    var path = attachment.SelectSingleNode("filePath")?.InnerText ?? string.Empty;
+                    var description = attachment.SelectSingleNode("description")?.InnerText;
+
+                    try
+                    {
+                        // We only support absolute paths since we dont lookup working directory here
+                        // any problem with path will throw an exception
+                        var fileUri = new Uri(path, UriKind.Absolute);
+                        attachmentSet.Attachments.Add(new UriDataAttachment(fileUri, description));
+                    }
+                    catch (UriFormatException ex)
+                    {
+                        _logger.Warning(string.Format("Ignoring attachment with path '{0}' due to problem with path: {1}", path, ex.Message));
```

Comment:
I'm part of in the middle here I think, I like new stuff, but one thing I like with NUnit is the backwards compatibility.  It makes it possible for industrial software, which I have been working on all my life, more or less, to use the stuff, they often cant upgrade as often as web sites or the like.   Now backwards compatibility have a limit though, at some points it becomes too hard and costly.
For this extension, it is a part of VS2017 (and former versions) but it has to run on the latest VS - and thus I see no problems staying with the latest standard there either, so I am happy about C# 7.  
One comment also on @CharliePoole's comment on string interpolation.   Contrary to Charlie, I really like it - I find the syntax much more readable, and I favour speed of reading.  My 5 cents  and my opinion of course. 
I like new things, but I don't like new things that don't really work, or is partly broken - and thus, I have built up a certain distaste for the .net core stuff.  I would love it to work out, but the fact it breaks in so many situations annoys me, and also the lack of tooling support it has.  I really hope it will improve rather soon.